website/addons/wiki/views.py

```diff
@@ -356,6 +363,44 @@ def project_wiki_edit_post(auth, wname, **kwargs):
         ret = {'status': 'success'}
     return ret, http.FOUND, None, redirect_url
 
+@must_be_valid_project  # injects node or project
+@must_have_permission('admin')
+@must_not_be_registration
+@must_have_addon('wiki', 'node')
+def edit_wiki_permissions(**kwargs):
+    node = kwargs['node'] or kwargs['project']
+    wiki_settings = node.get_addon('wiki')
+    permissions = kwargs.get('permissions')
+    auth = kwargs['auth']
+
+    if permissions is None:
+        raise HTTPError(http.BAD_REQUEST)
+
+    if wiki_settings is None:
+        raise HTTPError(http.BAD_REQUEST)
+
+    try:
+        edited = wiki_settings.set_editing(permissions, auth, node, True, True)
+    except AttributeError as e:
+        raise HTTPError(http.BAD_REQUEST, data=dict(
+            message_short=e.message
+        ))
+    else:
```

Comment:
Why the `else` block?

This is valid, but a little-used feature in Python - and one that's misunderstood often. Basically, the only time to use it is when you have something that needs to be done if the `try` block doesn't raise an exception, doesn't need to be done if it *does* raise an exception, and the exception is completely handled.

Since you're raising another exception in your handler, the `else` seems to serve no purpose. The `else` can just be removed and the code in that block can unindented one level.